### PR TITLE
clear fluid api: sigmoid_cross_entropy_with_logits

### PR DESCRIPTION
**Pull request by isky-cd**
### PR types

Others
### PR changes

APIs
### Describe

clear fluid api: sigmoid_cross_entropy_with_logits

### CODE REVIEW

**Review by sneaxiy — DISMISSED** [review]

File: python/paddle/nn/functional/loss.py [hunk 1]
```diff
@@ -4044,3 +4047,66 @@ def soft_margin_loss(input, label, reduction='mean', name=None):
         return paddle.mean(out, name=name)
     else:
         return out
+
+
+@templatedoc()
+def sigmoid_cross_entropy_with_logits(
```
Comment (jzhang533) [inline python/paddle/nn/functional/loss.py]: 我们确认是要暴露 `paddle.nn.functional.sigmoid_cross_entropy_with_logits` 这个API的是吗？

**Review by sneaxiy — DISMISSED** [review]

**Review by jzhang533 — DISMISSED** [review]
LGTM

File: python/paddle/nn/functional/loss.py (hunk 1 above)
Comment (isky-cd) [reply python/paddle/nn/functional/loss.py]: 已经隐藏

**Review by sneaxiy — DISMISSED** [review]

**Review by jzhang533 — DISMISSED** [review]
LGTM

File: python/paddle/fluid/tests/unittests/dygraph_to_static/test_word2vec.py [hunk 2]
```diff
@@ -263,7 +264,9 @@ def forward(self, center_words, target_words, label):
 
         pred = paddle.nn.functional.sigmoid(word_sim)
 
-        loss = fluid.layers.sigmoid_cross_entropy_with_logits(word_sim, label)
+        loss = paddle.nn.functional.loss.sigmoid_cross_entropy_with_logits(
```
Comment (XiaoguangHu01) [inline python/paddle/fluid/tests/unittests/dygraph_to_static/test_word2vec.py]: 这个不是标准的公开API的用法，建议换成binary_cross_entropy_with_logits函数进行测试。

File: python/paddle/nn/functional/loss.py (hunk 1 above)
Comment (XiaoguangHu01) [reply python/paddle/nn/functional/loss.py]: 这个函数可以去掉

File: python/paddle/nn/functional/loss.py [hunk 3]
```diff
@@ -4016,3 +4017,66 @@ def soft_margin_loss(input, label, reduction='mean', name=None):
...
+            import paddle
+
+            input = paddle.rand(shape=[10], dtype='float32')
+            label = paddle.rand(shape=[10], dtype='float32')
+            loss = paddle.nn.functional.loss.sigmoid_cross_entropy_with_logits(input, label,
+                                                            ignore_index=-1, normalize=True)
+            print(loss)
+    """
+
+    if in_dygraph_mode():
```
Comment (XiaoguangHu01) [inline python/paddle/nn/functional/loss.py]: 这段逻辑在binary_cross_entropy_with_logits这个函数里已经有了

File: python/paddle/nn/functional/loss.py [hunk 4]
```diff
@@ -4016,3 +4017,66 @@ def soft_margin_loss(input, label, reduction='mean', name=None):
...
+        return _C_ops.sigmoid_cross_entropy_with_logits(
+            x, label, normalize, int(ignore_index)
+        )
+    check_variable_and_dtype(
```
Comment (XiaoguangHu01) [inline python/paddle/nn/functional/loss.py]: 可以把下面这段逻辑，迁移到binary_cross_entropy_with_logits的静态图实现里

**Review by jzhang533 — DISMISSED** [review]
LGTM

**Review by sneaxiy — DISMISSED** [review]

**Review by XiaoguangHu01 — DISMISSED** [review]
LGTM

File: python/paddle/fluid/tests/unittests/test_sigmoid_cross_entropy_with_logits_op.py [hunk 5]
```diff
@@ -56,18 +49,17 @@ def setUp(self):
         self.outputs = {'Out': -term1 - term2}
 
     def test_check_output(self):
-        self.check_output(check_eager=True)
+        self.check_output(check_eager=False)
```
Comment (JiabinYang) [inline python/paddle/fluid/tests/unittests/test_sigmoid_cross_entropy_with_logits_op.py:52]: why this?

**Review by JiabinYang — COMMENTED** [review]
some comments

File: python/paddle/fluid/tests/unittests/test_sigmoid_cross_entropy_with_logits_op.py (hunk 5 above)
Comment (isky-cd) [reply python/paddle/fluid/tests/unittests/test_sigmoid_cross_entropy_with_logits_op.py:52]: 因为我们已经把sigmoid_cross_entropy_with_logits删除了 之前sigmoid_cross_entropy_with_logits的op测试会调用之前layers.sigmoid_cross_entropy_with_logits的python_api check_eager=False就是在测试时不调用layers.sigmoid_cross_entropy_with_logits的python_api

**Review by sneaxiy — APPROVED** [review]

**Review by jzhang533 — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM
